packages/core/src/services/commits/runDocumentAtCommit.ts

```diff
@@ -69,22 +104,24 @@ function enqueueEvent(
 async function iterate({
   chain,
   previousApiKey,
-  scope,
+  allApiKeys,
```

Comment:
Now instead of doing N SQL queries we do on query for all api keys in workspace which usually will be a small collection. From here we just check in a `Map` if the key is found by `name`